packages/shared/src/components/cards/ActionButtons.tsx

```diff
@@ -44,51 +53,58 @@ export default function ActionButtons({
         className,
       )}
     >
-      <QuaternaryButton
-        id={`post-${post.id}-upvote-btn`}
-        icon={<UpvoteIcon />}
-        buttonSize="small"
-        pressed={post.upvoted}
-        {...getTooltipProps(post.upvoted ? 'Remove upvote' : 'Upvote')}
-        onClick={() => onUpvoteClick?.(post, !post.upvoted)}
-        style={{ width: rem(78) }}
-        className="btn-tertiary-avocado"
-      >
-        <InteractionCounter value={post.numUpvotes > 0 && post.numUpvotes} />
-      </QuaternaryButton>
+      <Tooltip content={post.upvoted ? 'Remove upvote' : 'Upvote'}>
+        <QuaternaryButton
+          id={`post-${post.id}-upvote-btn`}
+          icon={<UpvoteIcon />}
+          buttonSize="small"
+          pressed={post.upvoted}
+          onClick={() => onUpvoteClick?.(post, !post.upvoted)}
+          style={{ width: rem(78) }}
+          className="btn-tertiary-avocado"
+        >
+          <InteractionCounter value={post.numUpvotes > 0 && post.numUpvotes} />
+        </QuaternaryButton>
+      </Tooltip>
+      {getShouldLoadTooltip() && (
+        <LazyTooltip content="Comments" reference={commentRef} />
+      )}
```

Comment:
Also with this one, why can't we use the normal tooltip method here?

Reply:
Sorry for not explaining the actual reason in the description I will update it with the details below:

`next/link` has an existing issue when wrapping a custom component, so when we use the tooltip as a child and wrap the actual content of the `next/link` it will not work. We have to use its reference to identify the trigger of the actual tooltip. Thankfully, the library we are using supports it out of the box.

https://github.com/vercel/next.js/issues/7915